src/main/java/xyz/nucleoid/extras/lobby/contributor/ContributorData.java

```diff
@@ -0,0 +1,148 @@
+            .setNameFormat("contributor-data-fetcher")
+            .setDaemon(true)
+            .build()
+    );
+
+    private static final String PEOPLE_KEY = "people";
+    private static final String NAME_KEY = "name";
+    private static final String SOCIALS_KEY = "socials";
+    private static final String MINECRAFT_KEY = "minecraft";
+    private static final String STATUE_NBT_KEY = "statue_nbt";
+
+    private static final Map<String, Contributor> CONTRIBUTORS = new HashMap<>();
+
+    // Methods for accessing data
+
+    public static Iterable<Entry<String, Contributor>> getContributors() {
+        return CONTRIBUTORS.entrySet();
+    }
+
+    public static Contributor getContributor(String id) {
+        return CONTRIBUTORS.get(id);
+    }
+
+    // Methods for reloading data
+
+    private static void addContributor(Entry<String, JsonElement> entry) {
+        try {
+            var contributorJson = entry.getValue().getAsJsonObject();
+
+            var name = contributorJson.get(NAME_KEY).getAsString();
+
+            var socials = contributorJson.getAsJsonObject(SOCIALS_KEY);
+            var minecraftUuidString = socials.get(MINECRAFT_KEY).getAsString();
+
+            var uuidInteger = new BigInteger(minecraftUuidString, 16);
+            var minecraftUuid = new UUID(uuidInteger.shiftRight(64).longValue(), uuidInteger.longValue());
+
+            var statueNbtJson = contributorJson.getAsJsonObject(STATUE_NBT_KEY);
+            var statueNbt = statueNbtJson == null ? null : JsonOps.INSTANCE.convertTo(NbtOps.INSTANCE, statueNbtJson);
+
+            var contributor = new Contributor(name, minecraftUuid, (NbtCompound) statueNbt);
+            CONTRIBUTORS.put(entry.getKey(), contributor);
+        } catch (Exception e) {
+            LOGGER.warn("Failed to parse contributor '{}'", entry.getKey(), e);
+        }
+    }
+
+    private static void refreshData() {
+        CONTRIBUTORS.clear();
+
+        try {
+            var config = NucleoidExtrasConfig.get();
+            if (config.contributorDataUrl() == null) return;
+
+            var url = new URL(config.contributorDataUrl());
+            var connection = (HttpsURLConnection) url.openConnection();
+
+            connection.setDoOutput(true);
+            connection.setRequestMethod("GET");
+            connection.setRequestProperty("Content-Type", "application/json");
+
+            try (var reader = new InputStreamReader(connection.getInputStream(), StandardCharsets.UTF_8)) {
+                var json = JsonHelper.deserialize(reader);
+                var people = json.getAsJsonObject(PEOPLE_KEY);
+
+                for (var entry : people.entrySet()) {
+                    addContributor(entry);
+                }
+            }
+        } catch (IOException e) {
+            LOGGER.error("Failed to fetch contributor data", e);
+        }
+    }
+
+    private static void refreshHolograms(MinecraftServer server) {
```

Comment:
Can this be run on the server thread? This seems scary

Reply:
Fixed in 8114a2000a3b9c30b06ee2f7b81a21c5ce15578f